client/jobs.lua

```diff
@@ -34,25 +33,23 @@ function CreateJobBlip() -- Used globally
     EndTextCommandSetBlipName(CurrentBlip)
 end
 
---- This will set the job as done and give a new location at the same time for you to continue the job and give you some time cut as a reward
-local function onBoxDone()
+local function onTaskDone()
     if math.random(1, 100) <= 50 then
         exports.qbx_core:Notify(locale("success.time_cut"))
         JailTime -= math.random(1, 2)
     end
 
-    if currentBox == #electricalBoxes then
-        electricalBoxes = newRandomArray(#Config.Jobs.electrician.locations)
-        currentBox = 1
+    if currentTask == #jobLocations then
+        jobLocations = newRandomArray(#Config.Jobs[prisonJob].locations)
+        currentTask = 1
         TriggerServerEvent('qbx_prison:server:completedJob')
     else
-        currentBox += 1
+        currentTask += 1
     end
 
     CreateJobBlip()
 end
 
---- This will be triggered once you interact with a job location to perform your job at
 local function startWork()
```

Comment:
You'd want to change the animation and locales of the progress bar depending on the job, no?